### ISSUE TITLE
Angular Keeps Compiling

### ISSUE THREAD

**Issue by irshadahmed** [angular/angular-cli#23348]
### Which @angular/* package(s) are the source of the bug?

compiler-cli, compiler, core

### Is this a regression?

No

### Description

I have setup a fresh Angular project. When I run using **ng serve** it keeps compiling non-stop without any real change in the source code.

<img width="500" alt="image" src="https://user-images.githubusercontent.com/8254178/173044009-f4c760d3-a6cd-4340-808a-b5362fb0ca4e.png">

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 13.3.7
Node: 16.15.1
Package Manager: npm 8.12.1
OS: win32 x64

Angular:
Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1303.7 (cli-only)
@angular-devkit/core         13.3.7 (cli-only)
@angular-devkit/schematics   13.3.7 (cli-only)
@schematics/angular          13.3.7 (cli-only)
```

### Anything else?

Please help me fix this problem, am struggling from past 1 week and nothing is helping out of Google as well.

**Comment by alan-agius4** [comment]
You can check which files have been modified/deleted by replacing liner 67 in `node_modules/@angular-devkit/build-angular/src/webpack/configs/common.js` with the below

```diff
-    const extraPlugins = [];
+    const extraPlugins = [
+      new (class {
+        apply(compiler) {
+          compiler.hooks.watchRun.tap(
+            "WatchRunPlugin",
+            ({ modifiedFiles, removedFiles }) => {
+              console.log({ modifiedFiles, removedFiles });
+            }
+          );
+        }
+      })()
+    ];
```

**Comment by irshadahmed** [comment]
> You can check which files have been modified/deleted by replacing liner 67 in `node_modules/@angular-devkit/build-angular/src/webpack/configs/common.js` with the below
> 
> ```diff
> -    const extraPlugins = [];
> +    const extraPlugins = [
> +      new (class {
> +        apply(compiler) {
> +          compiler.hooks.watchRun.tap(
> +            "WatchRunPlugin",
> +            ({ modifiedFiles, removedFiles }) => {
> +              console.log({ modifiedFiles, removedFiles });
> +            }
> +          );
> +        }
> +      })()
> +    ];
> ```

This is not helping identify the changes.
<img width="695" alt="image" src="https://user-images.githubusercontent.com/8254178/173090163-cd3f88c6-70f1-45d5-ab2b-e3e1b8abb4ed.png">

**Comment by alan-agius4** [comment]
Can you run `ng version` inside your project?

**Comment by irshadahmed** [comment]
> Can you run `ng version` inside your project?

**My actual project is in Angular 11.x. At first, I faced the compilation issue in my current project.** Then, I did setup a new Angular project with version 13.x. Surprising I faced the same compilation issue there as well. Hence I think, we need to fix this in both Angular 11.x & Angular 13.x

ng version - Current Project with Angular 11.x
<img width="392" alt="image" src="https://user-images.githubusercontent.com/8254178/173110011-529f10fd-0e45-4ced-a6b3-a49676422728.png">

ng version - New Project with Angular 13.x
<img width="392" alt="image" src="https://user-images.githubusercontent.com/8254178/173110166-e1ee03bb-5ced-4f2e-859f-094bb5c0e447.png">

**Comment by alan-agius4** [comment]
From both the screenshots above. Version 11 of Angular is used in both cases. In the 2nd case only Angular CLI is version 13. 

Please align all Angular packages to the same major version. 

Please also be aware that Angular 11 is no longer supported. Please see https://angular.io/guide/releases#support-policy-and-schedule

**Comment by irshadahmed** [comment]
> From both the screenshots above. Version 11 of Angular is used in both cases. In the 2nd case only Angular CLI is version 13.
> 
> Please align all Angular packages to the same major version.
> 
> Please also be aware that Angular 11 is no longer supported. Please see https://angular.io/guide/releases#support-policy-and-schedule

Extremely sorry. I didn't realize I pasted the same screenshot in both the cases. Here is the ng version for Angular 13.x project.
**ng version - New Project with Angular 13.x**
<img width="450" alt="image" src="https://user-images.githubusercontent.com/8254178/173169124-d30aadfe-7518-41f3-b30f-19e0c16789ad.png">

**Comment by alan-agius4** [comment]
Can you paste the provided snippet in the v13 project please?

**Comment by irshadahmed** [comment]
> Can you paste the provided snippet in the v13 project please?

I did that and here is what it displays in V13 project.
<img width="568" alt="image" src="https://user-images.githubusercontent.com/8254178/173180267-b6c17812-0bbc-4c78-9ec8-55a066426b00.png">

**Comment by alan-agius4** [comment]
Hi @irshadahmed, 

I was able to reproduce this and it seems that the problem is the special character `(` in the path. 

To be more precise this is exposing a bug in `glob-parent` package which is used by the `copy-webpack-plugin`.

https://github.com/webpack-contrib/copy-webpack-plugin/blob/b66cc95d1e67ca858c46d24fabadbb67f79ab009/src/index.js#L360

Example the below code
```
globParent('C:\\Users\\abc(a\\test-test\\src\\assets\\**\\*')
```

Will incorrectly return 
```
'C:\\Users'
```


Closing as this is not actionable from our end. I suggest to file an issue with `glob-parent`
